Sources/Leaf/Application+Leaf.swift

```diff
@@ -21,7 +21,7 @@ extension Application {
 
         public var renderer: LeafRenderer {
             var userInfo = self.userInfo
-            userInfo["application"] = self
```

Comment:
This is the crux of the issue. `self` here is of type `Application.Leaf`, but we want to assign an instance of `Application` for this key.